### DISCUSSION TITLE
Canadian postcodes seems slightly wrong

### DISCUSSION THREAD

**Discussion by felix-geovelo** [komoot/photon#798] [Q&A]
Hello !

It seems that Canadian postcodes are slightly wrong in the result given by photon.
Example:
With the input "5664 avenue Jeanne-d'Arc montreal" (https://photon.komoot.io/api?q=5664%20avenue%20Jeanne-d%27Arc)
with obtain the following result:
```{
  "features": [
    {
      "geometry": {
        "coordinates": [
          -73.57284865993546,
          45.55947595
        ],
        "type": "Point"
      },
      "type": "Feature",
      "properties": {
        "osm_id": 1187619697,
        "extent": [
          -73.5729229,
          45.5595384,
          -73.572774,
          45.5594139
        ],
        "country": "Canada",
        "city": "Montréal",
        "countrycode": "CA",
        "postcode": "H1X 1J5",
        "locality": "Rosemont",
        "county": "Agglomération de Montréal",
        "type": "house",
        "osm_type": "W",
        "osm_key": "building",
        "housenumber": "5664",
        "street": "Avenue Jeanne-D'arc",
        "district": "Rosemont–La Petite-Patrie",
        "osm_value": "apartments",
        "state": "Québec"
      }
    }
  ],
  "type": "FeatureCollection"
}
```

The result is globally good (position, etc), except the postcode, that give "H1X 1J5" instead of the real one beeing "H1X 2G3"
Is this a Photon issue or could it be a Nominatim issue ?
Thank you in advance

**Comment by lonvia** [comment]
This is am OpenStreetMap data issue. If you look at the [OSM object of the place](https://www.openstreetmap.org/way/1187619697), you see that no postcode is mapped. So Nominatim has to make an educated guess based on surrounding postcodes that are mapped. Photon just uses the postcodes that Nominatim has guessed.

**Reply by felix-geovelo** [reply]
Okay, so it would require a huge effort of mapping to fix that, thanks
